feat(display): port to v3 (#13318)
vuetifyjs/vuetify: packages/docs/src/pages/en/features/display-and-platform.md

---
meta:
  title: Display & Platform
  description: Access display viewport information using the Vuetify Breakpoint composable.
  keywords: breakpoints, grid breakpoints, platform details, screen size, display size
related:
  - /directives/resize/
  - /styles/display/
  - /styles/text-and-typography/
---

# Display & Platform

> The **Display** composable provides information on multiple aspects of the current device. This enables you to control various aspects of your application based upon the window size, device type, and SSR state. This composable works in conjunction with [grids](/components/grids/) and other responsive utility classes (e.g. [display](/styles/display/)).

Reviewer: ```suggestion
The **display** composable provides information on multiple aspects of the current device. This enables you to control various aspects of your application based upon the window size, device type, and SSR state. This composable works in conjunction with [grids](/components/grids/) and other responsive utility classes (e.g. [display](/styles/display/)).
```

          case 'lg': return 600
          case 'xl': return 800
        }
      },
    },
  }
</script>
```

### Interface

```ts
{
  // Breakpoints
  xs: boolean
  sm: boolean
  md: boolean
  lg: boolean
  xl: boolean
  smAndDown: boolean
  smAndUp: boolean
  mdAndDown: boolean
  mdAndUp: boolean
  lgAndDown: boolean
  lgAndUp: boolean

  // true if screen width < mobileBreakpoint
  mobile: boolean
  mobileBreakpoint: number | 'xs' | 'sm' | 'md' | 'lg' | 'xl'

  // Current breakpoint name (e.g. 'xs' | 'sm' | 'md' | 'lg' | 'xl')
  name: string

  // The current value of window.innerHeight and window.innerWidth
  height: number
  width: number

  // Device userAgent information
  platform: {
    android: boolean
    ios: boolean
    cordova: boolean
    electron: boolean
    chrome: boolean
    edge: boolean
    firefox: boolean
    opera: boolean
    win: boolean
    mac: boolean
    linux: boolean
    intersection: boolean
    touch: boolean
    ssr: boolean
  }

  // Browser scrollbar width
  scrollbarWidth: number

  // The values used to make Breakpoint calculations
  thresholds: {
    xs: number
    sm: number
    md: number
    lg: number
  }
}
```

> ## API

Reviewer: This section being moved to api generator?

Author: Oh I forgot about that. Will need some help implementing.

* **Type:** `object`
* **Default:**

  ```js
    {
      android: false,
      ios: false,
      cordova: false,
      electron: false,
      edge: false,
      firefox: false,
      opera: false,
      win: false,
      mac: false,
      linux: false,
      intersection: false,
      touch: false,
      ssr: false,
    }
  ```

### scrollbarWidth

This value is the width of the browser scrollbar.

* **Type:** `number`
* **Default:** `16`

### thresholds

This object is used for device viewport calculations.

* **Type:** `object`
* **Default:**

  ```js
    {
      xs: 600,
      sm: 960,
      md: 1280,
      lg: 1920,
      xl: 3840,
    }
  ```

* **Usage:**

  ```js
    // Composables
    import { useDisplay } from 'vuetify/lib/composables/display'

    export default {
      mounted () {
        const display = useDisplay()

        // display thresholds are not reactive
        // and do not need to use .value
        console.log(display.thresholds.md) // 1280
      }
    }
  ```

## Using Setup

> Setup is a new feature of Vue 3 that enables more control over the

Reviewer: ```suggestion
You can use the **display** composable alongside Vue 3's `setup` function to harness the power of the [Composition API](https://v3.vuejs.org/guide/composition-api-setup.html). In this example we show how to bind the **display** composable to the **fullscreen** property of `v-dialog`.
```

```html
<template>
  <v-dialog :fullscreen="display.mobile.value">
    ...
  </v-dialog>
</template>

<script>
  // Composables

>   import { useDisplay } from 'vuetify/lib/compsables/display'

Reviewer 2: ```suggestion
  import { useDisplay } from 'vuetify/lib/composables/display'
```

  export default {
    setup () {
      const display = useDisplay()

      return { display }
    },
  }
</script>
```

### JSX

> Using JSX with setup can accomplish the same thing as a template. The following example shows how to bind the same **fullscreen** property, but using JSX:

Reviewer: ```suggestion
Using JSX with the `setup` function you can accomplish the same thing as a template. The following example shows how to bind the **display** composable to the same **fullscreen** property of `v-dialog`, but using JSX:
```

Reviewer 3: Not sure if this section should even be here? Why would display composable specifically get example showing JSX syntax, when it's not used anywhere else in docs?

Author: I added it because the intention was to add it in other areas as a separate PR. However, it probably should be done as a separate PR for everything. Will remove this section.

```tsx
// Composables

> import { useDisplay } from 'vuetify/lib/compsables/display'

Reviewer 2: ```suggestion
import { useDisplay } from 'vuetify/lib/composables/display'
```

export default {
  setup () {
    const display = useDisplay()

    return () => {
      return (
        <v-dialog fullscreen={display.mobile.value}>
          ...
        </v-dialog>
      )
    }
  },
}
```

### Breakpoint conditionals

> The breakpoint and conditional values return a `boolean` that is derived from the current viewport size. Additionally, the breakpoint composable mimics the [Vuetify Grid](/components/grids) naming conventions and has access to properties such as `xlOnly`, `xsOnly`, `mdAndDown`, and others. In the following example we use the **setup** function to pass the **xs** and **mdAndUp** values to our template:

Reviewer: ```suggestion
Breakpoint and conditional values return a `boolean` that is derived from the current viewport size. Additionally, the **breakpoint** composable mimics the [Vuetify Grid](/components/grids) naming conventions and has access to properties such as **xlOnly**, **xsOnly**, **mdAndDown**, and many others. In the following example we use the `setup` function to pass the _xs_ and _mdAndUp_ values to our template:
```

```html
<!-- Vue Component -->

<template>
  <v-sheet
    :min-height="mdAndUp.value ? 300 : '20vh'"
    :rounded="xs.value"
  >
    ...
  </v-sheet>
</template>

<script>
  // Composables
  import { useDisplay } from 'vuetify/lib/composables/display'

  setup () {
    // Destructure only the keys we want to use
    const { xs, mdAndUp } = useDisplay()

    return { xs, mdAndUp }
  }
</script>
```

> Using the _dynamic_ display values, we are able to adjust the minimum height of [v-sheet](/components/sheets/) to **300** when on the _medium_ or greater breakpoint and only show rounded corners on _extra small_ screens:

Reviewer: ```suggestion
Using the _dynamic_ display values, we are able to adjust the minimum height of [v-sheet](/components/sheets/) to `300` when on the _medium_ breakpoint or greater and only show rounded corners on _extra small_ screens:
```

## Component Mobile Breakpoints

> Some components within Vuetify have a **mobile-breakpoint** property which allows you to override the default value. These components reference the global [mobileBreakpoint](#mobile-breakpoint) value that is generated at runtime using the provided options in the `vuetify.js` file. By default, **mobileBreakpoint** is set to **md**, which means that if the window is less than _1280_ width,—the default value for the **md** [threshold](#thresholds)—the display composable will update its **mobile** value to `true`.

Reviewer: ```suggestion
Some components within Vuetify have a **mobile-breakpoint** property which allows you to override the default value. These components reference the global [mobileBreakpoint](#mobile-breakpoint) value that is generated at runtime using the provided options in the `vuetify.js` file. By default, **mobileBreakpoint** is set to `md`, which means that if the window is less than _1280_ pixels in width (which is the default value for the **md** [threshold](#thresholds)), then the **display** composable will update its **mobile** value to `true`.
```

> For example, the [v-banner](/components/banners/) component implements different styling based upon the value of **mobile** on the display composable . In the following example, The first banner uses the global mobile-breakpoint value of **'md'** while the second overrides this default with _580_. Assuming a screen width of _1024px_, the second banner would not convert to its mobile state:

Reviewer: ```suggestion
For example, the [v-banner](/components/banners/) component implements different styling based upon the value of **mobile** on the **display** composable. In the following example, The first banner uses the global **mobile-breakpoint** value of `md` while the second overrides this default with `580`. If the screen width is 1024 pixels, the second banner would not convert into its mobile state:
```
